src/cmap/status.rs

```diff
@@ -0,0 +1,90 @@
+use crate::error::Error;
+
+/// Struct used to track the latest status of the pool.
+#[derive(Clone, Debug)]
+struct PoolStatus {
+    /// The current generation of the pool.
+    generation: u32,
+
+    /// The last connection establishment error that caused the pool to be cleared,
+    /// if that was the cause of the clearing.
+    establishment_error: Option<Error>,
+}
+
+impl Default for PoolStatus {
+    fn default() -> Self {
+        PoolStatus {
+            generation: 0,
+            establishment_error: None,
+        }
+    }
+}
+
+/// Create a channel for publishing and receiving updates to the pool's generation.
+pub(super) fn channel() -> (PoolGenerationPublisher, PoolGenerationSubscriber) {
```

Comment:
This channel is currently used by the monitor to detect connection establishment errors, but it will also be used in the future to get the current generation as part of implementing the rest of RUST-355.

Happy to discuss design / naming options for the types in this file.